test/com/rpl/specter/core_test.cljc

```diff
@@ -960,7 +960,26 @@
   (is (= [[] [2] [4 6]]
          (select
            [(s/continuous-subseqs number?) (s/filterer even?)]
-           [1 "a" "b" 2 3 "c" 4 5 6 "d" "e" "f"]))))
+           [1 "a" "b" 2 3 "c" 4 5 6 "d" "e" "f"])))
+  (defn- make-bounds-pred-fn [start end]
```

Comment:
Couple comments here:

- The defn should be a top-level form
- I'm struggling to understand this function. Shouldn't it be returning true or false? Why does it return `start` and `end`  in some cases. Is `prev` the previous element or true/false for the run of this function on the previous element?

Reply:
Yeah, it's a little dicey.  Basically, it's trying to have the "previous" value do double duty: both signaling that the item should be included in a selected subsequence (i.e. `true`), but also carrying the previous boundary keyword (if applicable), without indicating that keyword should be included in a subsequence.  So, more or less the same concept as outlined in the issue #236 description.

I tried changing the predicate around so that it returns a vector of both of these separately, but it seems to get a bit gnarly trying to handle that in the transducer:

```    (let [index-results (into [] (subseq-pred-fn-transducer p) aseq)]
      (map last (filter (comp true? first) index-results)))```

I suppose the caller could be asked to supply an additional function to `subseq-pred-fn` that controls how the truthy piece is pulled out of the results for the purpose of building the actual subsequence indeces?